### PR TITLE
disable prefer destructuring for arrays

### PR DESCRIPTION
**Pull request by airondumael**
disable `prefer-destructuring` for arrays

### CODE REVIEW

File: eslint.js [hunk 1]
```diff
@@ -336,7 +336,7 @@ module.exports = {
         'prefer-arrow-callback': 'error',
         'prefer-const': 'error',
         'prefer-destructuring': ['error', {
```
Comment (DaleLJefferson) [inline eslint.js]: I thought we would just turn the whole rule off, I think object has the same problems

**Comment by fjmorant** [comment]
Could you explain why we are disabling this rule? @airondumael

File: eslint.js (hunk 1 above)
Comment (airondumael) [reply eslint.js]: I thought it's just array. will update the PR then

**Comment by airondumael** [comment]
@javierM84 ,
It's because if we follow this rule the generated code sometimes have auto semicolon insertion issue.
It was discussed on our development channel https://rnfdigital.slack.com/archives/development/p1486393806001567

**Review by fjmorant — APPROVED** [review]
👍

**Pull request merged**